src/meltano/cli/add.py

```diff
@@ -89,27 +88,24 @@ def add(  # noqa: WPS238
         inherit_from = plugin_names[0]
         plugin_names = [as_name]
 
-    plugins_service = ProjectPluginsService(project)
-
-    if flags["custom"]:
-        if plugin_type in {
-            PluginType.TRANSFORMS,
-            PluginType.ORCHESTRATORS,
-        }:
-            tracker.track_command_event(CliEvent.aborted)
-            raise CliError(f"--custom is not supported for {plugin_type}")
+    if flags["custom"] and plugin_type in {
+        PluginType.TRANSFORMS,
+        PluginType.ORCHESTRATORS,
+    }:
+        tracker.track_command_event(CliEvent.aborted)
+        raise CliError(f"--custom is not supported for {plugin_type}")
 
     plugin_refs = [
         PluginRef(plugin_type=plugin_type, name=name) for name in plugin_names
     ]
     dependencies_met, err = check_dependencies_met(
-        plugin_refs=plugin_refs, plugins_service=plugins_service
+        plugin_refs=plugin_refs, plugins_service=project.plugins
     )
     if not dependencies_met:
         tracker.track_command_event(CliEvent.aborted)
         raise CliError(f"Failed to install plugin(s): {err}")
 
-    add_service = ProjectAddService(project, plugins_service=plugins_service)
+    add_service = ProjectAddService(project)
```

Comment:
Yes, it can, along with several other similar "service" classes that primarily depend on the `Project` instance or something that is now (or will be) part of the `Project` instance.

I'll probably do another few refactors along this line in the future, but I wanted to pause here to avoid further increasing the scope, and to deliver a stable increment.